pkg/kubectl/resource/mapper.go

```diff
@@ -44,17 +34,27 @@ type Mapper struct {
 	runtime.Decoder
 }
 
+// AcceptUnrecognizedObjects will return a mapper that will tolerate objects
+// that are not recognized by the RESTMapper, returning mappings that can
+// perform minimal transformation. Allows working in disconnected mode, or with
+// objects that the server does not recognize. Returned resource.Info objects
+// may have empty resource fields and nil clients.
+func (m *Mapper) AcceptUnrecognizedObjects() *Mapper {
+	copied := *m
+	copied.RESTMapper = NewRelaxedRESTMapper(m.RESTMapper)
+	copied.ClientMapper = NewRelaxedClientMapper(m.ClientMapper)
+	return &copied
+}
+
 // InfoForData creates an Info object for the given data. An error is returned
 // if any of the decoding or client lookup steps fail. Name and namespace will be
 // set into Info if the mapping's MetadataAccessor can retrieve them.
 func (m *Mapper) InfoForData(data []byte, source string) (*Info, error) {
-	versions := &runtime.VersionedObjects{}
```

Comment:
Is kubectl the only consumer of `VersionedObjects{}` do we also need to remove [here](https://github.com/kubernetes/kubernetes/blame/3df3c580b7a318c8fd166296b61d891f6c0359fa/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go#L93) and append logic [there](https://github.com/kubernetes/kubernetes/blame/3df3c580b7a318c8fd166296b61d891f6c0359fa/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go#L135-L138) ?

Reply:
Sorry, i think i forget the compatible problem...please omit

Reply:
I will be removing VersionedObjects in a future PR.